application/helpers/update/DatabaseUpdateBase.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace LimeSurvey\Helpers\Update;
+
+use CDbConnection;
+use Exception;
+use Throwable;
+
+/**
+ * Base class for database migration, inspired by Yii.
+ * See more info in README file in same folder.
+ */
+abstract class DatabaseUpdateBase
+{
+    /** @var CDbConnection */
+    protected $db;
+
+    /** @var int */
+    protected $newVersion;
+
+    /** @var string */
+    protected $options;
+
+    /**
+     * @param CDbConnection $connection
+     * @param string $options Specific database options like ENGINE=INNODB etc
+     */
+    public function __construct(CDbConnection $connection, $options)
+    {
+        $this->db = $connection;
+        // Database version is part of class and file name, e.g. Update_123.
+        $this->newVersion = $this->getVersion();
+        $this->options = $options;
+    }
+
+    /**
+     * Runs up() wrapped in a transaction.
+     * Will rollback transaction and re-throw exception at failure.
+     *
+     * @return void
+     * @throws Throwable
+     */
+    public function safeUp()
+    {
+        $transaction = $this->db->beginTransaction();
+        try {
+            $this->up();
+            $this->updateVersion();
+            $transaction->commit();
+        } catch (Throwable $e) {
+            $transaction->rollback();
+            throw $e;
+        }
+    }
+
+    /**
+     * Sets DBVersion in settings global to $this->newVersion
+     * Last thing that happens before transaction commit in every update.
+     */
+    private function updateVersion()
+    {
+        $this->db
+            ->createCommand()
+            ->update('{{settings_global}}', ['stg_value' => $this->newVersion], "stg_name='DBVersion'");
+    }
+
+    /**
+     * Get db version number based on class name, e.g. 123 for Update_123
```

Comment:
Why not for `Update_` ?

Reply:
Well. Maybe. Two things to think of:

1) Future proof: at some point we'll have Update_XXXX
2) Hacker proof? We might want to introduce timestamps or "Update_123b" or other things. That can be added later, of course.

But sure, you can throw exception if first part of explode() is not "Update".

Reply:
Unsure here … but think it's always best to start by more restriction than needed. And remove restriction after.

I read about number or date or … in mantis. No good advice …

Reply:
Pushed a check for $nameParts[0]

Reply:
But i din't ask for ;)

Reply:
Probably a good idea.